sdks/go/pkg/beam/core/graph/fn_test.go

```diff
@@ -470,6 +542,169 @@ func (fn *BadDoFnAmbiguousSideInput) StartBundle(bool) {
 func (fn *BadDoFnAmbiguousSideInput) FinishBundle(bool) {
 }
 
+// Examples of correct SplittableDoFn signatures
+
+type RestT struct{}
+
+type GoodSdf struct {
```

Comment:
Shouldn't the ProcessElement method be adjusted here for the Restriction tracker parameter for a "GoodSdf" ? 

I guess technically if we don't care about dynamic splitting at all, it should be allowed. It's not unreasonable to prevent it until we have the right idea how to do that, but if so, lets put a TODO in here somewhere to make the intent explicit.

Reply:
I was just skipping it because this commit hasn't added restriction trackers yet, but adding a TODO to remind me to change it later is a good idea. Done.

Reply:
Makes sense! Thanks!